### ISSUE TITLE
[feature] tools.PkgConfig should honor $PKG_CONFIG.

### ISSUE THREAD

**Issue by coldtobi** [conan-io/conan#8591]
(My Version Info: Conan 1.33, pip-installed on Debian unstable)

Many thanks for providing conan! I might have a suggestion how to improve the tools.PkgConfig integration, as it is currently not behaving as expected-by-convention: It is ignoring the PKG_CONFIG env…

The autotools documentation mentions this in [1] that one can specify the pkg-config tool to be used by setting env PKG_CONFIG. It's also mentioned in the manpage of pkg-config (autotools macro section). Also, cmake mentions it in
[2] and my experiments shows that indeed autotools and cmake honors it. Calling ./configure, on projects needing it, without pkg-config being present tells you to set PKG_CONFIG to point to the executable [3].

Unfortunately conan's tools.PkgConfig does not honor it; so the "wrong" pkg-config will be called. Of course, tools.PkgConfig can be hinted explicitly, but this is cumbersome to do so in every recipe if not using the documented PKG_CONFIG way.

My use case: I'm running into a pkg-config bug [4]  Until that one is fixed, I need a local wrapper to work around the problem and the script is provided by an conan package and overrides the default pkg-config (but still needs to call it.)

This would make it more easy to check other pkg-config implementations, e.g pkgconf (which did not help with my issue, as it has the same issue [5]…)

[1] https://autotools.io/pkgconfig/cross-compiling.html  -- Section 5.2. Tool Calling Conventions
Quote:  "This is supported by the autoconf macros provided by the package, as they all respect $PKG_CONFIG if set in the environment,"
[2] https://cmake.org/cmake/help/latest/module/FindPkgConfig.html#variables-affecting-behavior
Quote: The PKG_CONFIG environment variable can be used as a hint.
[3] https://autotools.io/pkgconfig/pkg_check_modules.html -- see the example output in 3.4. Optional Modules section or
     another IRL example at https://bugs.debian.org/528169
[4] https://gitlab.freedesktop.org/pkg-config/pkg-config/-/issues/52
[5] https://github.com/pkgconf/pkgconf/issues/213

**Comment by danimtb** [comment]
Hi @coldtobi and thanks a lot for the detailed report.

Indeed, the PKG_CONFIG var seems to be a standard supported by many tools and at first sight your request seems legit. I have opened a pull request implementing this at #8601 but it is something preliminary as we generally we are not adding more features to Conan 1.X (trying to focus on Conan 2 for new features) soi I cannot tell if it will be included in next releases. Anyway, you can give it a try from my branch if you wish 😄

**Comment by coldtobi** [comment]
@danimtb I've just tested your PR and can cofirm that it works! Thanks for the quick reaction! 

I've been reading the PR, but please note that I'm not experienced in python at all, so this might be /me applying C++ logic to Python and my comment might be plain wrong:
 
`        self.pkg_config_executable = pkg_config_env or pkg_config_executable `

Does that mean that the PKG_CONFIG get preference over pkg_config_executable? (My bikesheed color would be the opposite, as this would always override the parameter given to the function with the env, and it would no longer be possible to do an local override with the function parameter…)

**Comment by danimtb** [comment]
@coldtobi yeah, that is the question we got as well. Probably we will have to go with the `global.conf` way https://docs.conan.io/en/latest/reference/config_files/global_conf.html instead and maybe provide a syntax there so the user is able to specify the $PKG_CONFIG env var value to have full flexibility
